lockc/src/bpf/lockc.bpf.c

```diff
@@ -35,62 +35,59 @@
  * Return: 0 if all operations sucessful, otherwise an error code.
  */
 static __always_inline int handle_new_process(struct task_struct *parent,
-					      struct task_struct *child)
-{
-	int err;
-	pid_t pid = BPF_CORE_READ(child, pid);
-	pid_t ppid = BPF_CORE_READ(parent, pid);
-
-	/* Check if parent process is containerized. */
-	struct process *parent_lookup = bpf_map_lookup_elem(&processes, &ppid);
-	if (!parent_lookup) {
-		/* If not, check whether it's a container runtime process. */
-		// const char *comm = BPF_CORE_READ(child, comm);
-		// u32 runtime_key = hash(comm, TASK_COMM_LEN);
-		// u32 *runtime_lookup = bpf_map_lookup_elem(&runtimes,
-		// 					  &runtime_key);
-		// if (runtime_lookup) {
-		// 	/*
-		// 	 * If yes, it means that's an unwrapped container
-		// 	 * runtime process. Deny it.
-		// 	 */
-		// 	bpf_printk("deny: unwrapped runtime process %d: %s\n",
-		// 		   pid,
-		// 		   BPF_CORE_READ(child, comm));
-		// 	return -EPERM;
-		// }
-		return 0;
-	}
-
-	/* Skip registration if process entry already exists. */
-	struct process *v = bpf_map_lookup_elem(&processes, &pid);
-	if (v != NULL)
-		return 0;
-
-	bpf_printk("found parent containerized process: %d\n", ppid);
-	bpf_printk("comm: %s\n", BPF_CORE_READ(child, comm));
-
-	u32 container_id = parent_lookup->container_id;
-	u32 *container_lookup = bpf_map_lookup_elem(&containers, &container_id);
-	if (!container_lookup) {
-		/* Shouldn't happen */
-		bpf_printk("error: handle_new_process: cound not find a "
-			   "container for a registered process %d, "
-			   "container id: %d\n",
-			   pid, container_id);
-		return -EPERM;
-	}
-
-	struct process new_p = {
-		.container_id = container_id
-	};
-
-	bpf_printk("adding containerized process: %d\n", pid);
-	err = bpf_map_update_elem(&processes, &pid, &new_p, 0);
-	if (err < 0)
-		return err;
-
-	return 0;
+                                              struct task_struct *child) {
+  int err;
+  pid_t pid = BPF_CORE_READ(child, pid);
+  pid_t ppid = BPF_CORE_READ(parent, pid);
+
+  /* Check if parent process is containerized. */
+  struct process *parent_lookup = bpf_map_lookup_elem(&processes, &ppid);
+  if (!parent_lookup) {
+    /* If not, check whether it's a container runtime process. */
+    // const char *comm = BPF_CORE_READ(child, comm);
+    // u32 runtime_key = hash(comm, TASK_COMM_LEN);
+    // u32 *runtime_lookup = bpf_map_lookup_elem(&runtimes,
+    // 					  &runtime_key);
+    // if (runtime_lookup) {
+    // 	/*
+    // 	 * If yes, it means that's an unwrapped container
+    // 	 * runtime process. Deny it.
+    // 	 */
+    // 	bpf_printk("deny: unwrapped runtime process %d: %s\n",
+    // 		   pid,
+    // 		   BPF_CORE_READ(child, comm));
+    // 	return -EPERM;
+    // }
```

Comment:
nitpicking, if this is commented, this is not needed. Given this is kept inside of git's history, I would remove all these lines

Reply:
It's actually not needed at all, since runtime detection is done in fanotify. I completely forgot about that code. Will remove in a separate PR.

Reply:
https://github.com/rancher-sandbox/lockc/pull/148